### DISCUSSION TITLE
not getting data over RS-485

### DISCUSSION THREAD

**Discussion by raymop** [tagyoureit/nodejs-poolController#453] [RS485] [answered]
I am setting up the pool controller for the second time.  I had it all running quite nicely in 2021.
But, I am having trouble with the RS-485 signals.
I am using the same A/B wires from last year. One is from a Pentair Intelliflo pump, and the other is from an Aqua-Rite salt water generator.  Both were sending signals last year.
This week, I am getting no data or signals over either set of wires.
I tried using the same RS-485 to USB adapter as last year.
I bought. new adapter this week. Neither show any TX/RX red blips. And the power light is red on the adapter.
I reversed the wiring to no effect.
I get /dev/ttyUSB0.
I tried  "od -x < /dev/ttyUSB0" with no results.
If I unplug the RS-485 adapter from the USB port on the RPI 4 while od -x < /dev/ttyUSB0, and I get "00000000" and then a prompt.

The wire pairs are coming through conduit into a weather proof electric enclosure.

Thoughts?????!!!!

**Comment by tagyoureit** [comment, marked as answer]
If you are trying to exclusively use this app and there is no OCP (outdoor control panel) in play, then the 'od -x ...' won't show anything.  The pump and chlorinator will only respond when sent the right combination of that we send it.

If you set this up a year+ ago, things may have changed a bit.  You now need to go and manually add your pump and chlorinator.  We don't automatically detect them anymore.

Also, try just starting with the pump.  The Aquarite cells need 10v applied to it on it will not talk back.  

See https://github.com/tagyoureit/nodejs-poolController/discussions/415.

**Reply by raymop** [reply]
Thanks.  I seem to have a connection now.
But, when I attach the RS-485 to the RPI4, my pool pump stops and the pump display reads “Display not active”
I have to power down the circuit, then restart the pump, and it recovers.
And when I e-connect RS-485 I get the same behavior.
Here are the replay files, which include screen shots of the DashPanel.
Would love your troubleshooting!!!!
[replay-4.zip](https://github.com/tagyoureit/nodejs-poolController/files/8788051/replay-4.zip)

**Reply by raymop** [reply]
My bad.  I figured it out.  Had to add a circuit called pool.

**Comment by ddyer8** [comment]
I'm also having trouble configuring my SuperFlo VST.  It's the new version s manufactured after 10/15/20, so I'm configuring as Intelliflo VS.

I'm attempting to use the RS-485 port on the SM megabas card.  I have only yellow/green wired to the pump on the data cable.  I made my own data cable because I wasn't able to source the digital control kit.  I believe I have the wires on the correct pins (2,3).  The pump flashes "Auto" on the display, but in the njsPC logs I get:
3|njsPC  | Processing schedule 1 - Pool : ShouldBeOn: true ManualPriorityActive: false Running: true Suspended: false Resumed: false
3|njsPC  | [5/22/2022, 8:39:45 PM] warn: Message aborted after 3 attempt(s): 165,0,96,33,5,1,6,1,50
3|njsPC  | [5/22/2022, 8:39:45 PM] error: Error sending setPumpManual for Pump: Message aborted after 3 attempt(s): 165,0,96,33,5,1,6,1,50
3|njsPC  | Processing schedule 1 - Pool : ShouldBeOn: true ManualPriorityActive: false Running: true Suspended: false Resumed: false

Any troubleshooting steps would be greatly appreciated!!!

**Reply by tagyoureit** [reply]
I moved this to it's own issue.  https://github.com/tagyoureit/nodejs-poolController/issues/457
